### PR TITLE
[oracle_weblogic] Add Integration Package with Managed Server Logs Data Stream

### PR DESCRIPTION
**Pull request by kush-elastic**
- Enhancement

## What does this PR do?

- Generated the skeleton of Oracle WebLogic integration package.
- Added 1 data stream (Managed Server Logs).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/oracle_weblogic directory.
- Run the following command to run tests.
`elastic-package test`

## Related issues

- Relates #2823
- Relates #2910
- Relates #3066
- Relates #3068
- Relates #3069
- Relates #3121 
- Relates #3189

## Screenshots

oracle_weblogic  Managed Server Dashboard

### CODE REVIEW

File: packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    - log
+    ignore_failure: true
+- rename:
+    field: message
+    target_field: event.original
+    ignore_missing: true
+    ignore_failure: true
+- dissect:
+    field: event.original
+    pattern: "####<%{@timestamp}> <%{log.level}> <%{oracle_weblogic.managed_server.subsystem}> <%{oracle_weblogic.managed_server.machine_name}>
```
Comment (agithomas) [inline packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml:41]: Please check the order if the order is Server Name , Machine Name, Thread Id

File: packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+description: Pipeline for processing Weblogic logs
```
Comment (agithomas) [inline packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml:2]: Please correct as Pipeline for processing WebLogic managed server logs

File: packages/oracle_weblogic/data_stream/managed_server/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  - input: logfile
+    vars:
+      - name: paths
+        type: text
+        title: Paths
+        multi: true
+        required: true
+        show_user: true
+        default:
+          - /tmp/service_logs/managed-logs/ms-1.log
```
Comment (agithomas) [inline packages/oracle_weblogic/data_stream/managed_server/manifest.yml:13]: Please correct the default path location

File: packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (kush-elastic) [reply packages/oracle_weblogic/data_stream/managed_server/elasticsearch/ingest_pipeline/default.yml:41]: The order will be Subsystem, Machine Name, Server Name, and Thread ID.
```
<May 2, 2022 12:16:27,431 PM GMT> <Info> <WorkManager> <managed_machine> <ms-1> <Timer-2> <<WLS Kernel>> <> <a3f7cb90-aee9-47d4-8977-6d437ae89137-0000000c> <1651493787431> <[severity-value: 64] [rid: 0] [partition-id: 0] [partition-name: DOMAIN] > <BEA-002959> <Self-tuning thread pool contains 2 running threads, 0 idle threads, and 12 standby threads>
```
WorkManager is Subsystem
managed_machine is Machine Name
ms-1 is Server Name
Timer-2 is Thread ID

**Review by agithomas — COMMENTED** [review]
Change looks good to me.

**Review by yug-elastic — COMMENTED** [review]
PR looks good to me, I checked the logs for the failure.
```
Error: error running package system tests: could not complete test run: could not setup service: removing service logs failed: removing resource failed (path: /var/lib/jenkins/workspace/est-manager_integrations_PR-3067/.elastic-package/tmp/service_logs/adminserver): unlinkat /var/lib/jenkins/workspace/est-manager_integrations_PR-3067/.elastic-package/tmp/service_logs/adminserver/domain1.log: permission denied
```
Looks like you'll need to fix something with the permissions there.

File: packages/oracle_weblogic/_dev/deploy/docker/oracle_weblogic/container-scripts/startAdminServer.sh [hunk 4]
```diff
@@ -110,6 +110,9 @@ check_wls "started" localhost ${ADMIN_PORT} 2
 
 /u01/oracle/generateAccess.sh &
 
+# Copy 'AdminServer' logfiles('admin-server.log','domain1.log' and 'access.log') for docker binding
```
Comment (yug-elastic) [inline packages/oracle_weblogic/_dev/deploy/docker/oracle_weblogic/container-scripts/startAdminServer.sh:113]: nit:
```suggestion
# Copy 'AdminServer' logfiles ('admin-server.log', 'domain1.log' and 'access.log') for Docker binding
```

File: packages/oracle_weblogic/_dev/deploy/docker/oracle_weblogic/container-scripts/startManagedServer.sh [hunk 5]
```diff
@@ -63,6 +63,10 @@ ADMIN_SERVER_URL="http://${ADMIN_HOST}:${ADMIN_PORT}"
 if [ "${SSL_ENABLED}" = "true" ]; then
   ADMIN_SERVER_URL="https://${ADMIN_HOST}:${ADMIN_SERVER_SSL_PORT}" 
 fi
+
+# Copy 'ManagedServer' logfile('managed-server1.log') for docker binding
```
Comment (yug-elastic) [inline packages/oracle_weblogic/_dev/deploy/docker/oracle_weblogic/container-scripts/startManagedServer.sh:67]: nit:
```suggestion
# Copy 'ManagedServer' logfile ('managed-server1.log') for Docker binding
```

**Review by yug-elastic — APPROVED** [review]
Just a couple of nits in comments to adjust, LGTM!

**Review by agithomas — APPROVED** [review]
Looks good to me.

**Pull request merged**